### ISSUE TITLE
Appium Desktop 1.3.0 - Error: Unable to launch WebDriverAgent because of xcodebuild failure: ENOENT: no such file or directory

### ISSUE THREAD

**Issue by jgbaylon** [appium/appium-desktop#389]
## Appium or Appium Desktop?

Appium Desktop 1.3.0

## The problem

When I try to use the inspector, I get the following error.

``
[MJSONWP] Encountered internal error running command: Error: Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
    at XCUITestDriver.quitAndUninstall$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/driver.js:375:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at 
``

## Environment

- I am running Appium Desktop version  1.3.0.
- I am on (pick one):
    - [X] Mac

**Comment by imurchie** [comment]
Please upgrade to Appium-Desktop version 1.3.1 https://github.com/appium/appium-desktop/releases/tag/v1.3.1

**Comment by jgbaylon** [comment]
@imurchie Thanks. I will try and report back.

**Comment by yotsubakoiwai** [comment]
@imurchie Hi Isaac, I just upgraded to Appium Desktop 1.3.2 with Appium v1.7.2, and I am encountering this error on MacOS High Sierra 10.13.3

When I was using Appium Desktop 1.3.1, the WebDriverAgent was installing on the iPhone simulator, but upon upgrading to the latest version, it started to fail.

Here's the logs:

> [Appium] Welcome to Appium v1.7.2
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"platformName":"iOS","deviceName":"iPhone 6s","app":"/Users/doomhammer/Downloads/tableAndPicker.app","automationName":"XCUITest","udid":"8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8"}}
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"platformName":"iOS","deviceName":"iPhone 6s","app":"/Users/doomhammer/Downloads/tableAndPicker.app","automationName":"XCUITest","udid":"8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8"},null,null]
[BaseDriver] Event 'newSessionRequested' logged at 1517882764684 (10:06:04 GMT+0800 (+08))
[Appium] Creating new XCUITestDriver (v2.67.0) session
[Appium] Capabilities:
[Appium]   platformName: iOS
[Appium]   deviceName: iPhone 6s
[Appium]   app: /Users/doomhammer/Downloads/tableAndPicker.app
[Appium]   automationName: XCUITest
[Appium]   udid: 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8
[BaseDriver] 
[BaseDriver] W3C capabilities [] and MJSONWP desired capabilities  were provided
[BaseDriver] Expected W3C "capabilities" to be a JSON Object but was provided with: []
[BaseDriver] Falling back to MJSONWP desired capabilities
[BaseDriver] Session created with session id: bbaebf3d-7bba-4295-a682-a89a9eca9b36
[XCUITest] Current user: 'doomhammer'
[XCUITest] Current version of libimobiledevice: stable 1.2.0 (bottled), HEAD
[XCUITest] Xcode version set to '9.2' (tools v9.2.0.0.1.1510905681)
[XCUITest] iOS SDK Version set to '11.2'
[BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1517882766389 (10:06:06 GMT+0800 (+08))
[iOSSim] Constructing iOS simulator for Xcode version 9.2 with udid '8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8'
[XCUITest] Determining device to run tests on: udid: '8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8', real device: false
[XCUITest] No platformVersion specified. Using device version: '11.2'
[BaseDriver] Using local app '/Users/doomhammer/Downloads/tableAndPicker.app'
[BaseDriver] Event 'appConfigured' logged at 1517882766890 (10:06:06 GMT+0800 (+08))
[XCUITest] Checking whether app '/Users/doomhammer/Downloads/tableAndPicker.app' is actually present on file system
[XCUITest] App is present
[iOS] Getting bundle ID from app '/Users/doomhammer/Downloads/tableAndPicker.app': 'com.chopsosaurus.tableAndPicker'
[BaseDriver] Event 'resetStarted' logged at 1517882766894 (10:06:06 GMT+0800 (+08))
[XCUITest] Not scrubbing third party app in anticipation of uninstall
[BaseDriver] Event 'resetComplete' logged at 1517882767172 (10:06:07 GMT+0800 (+08))
[XCUITest] Starting log capture for iOS Simulator with udid '8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8', using 'xcrun simctl spawn 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8 log stream --style compact'
[BaseDriver] Event 'logCaptureStarted' logged at 1517882767700 (10:06:07 GMT+0800 (+08))
[XCUITest] Setting up simulator
[iOS] No reason to set locale
[iOS] No iOS / app preferences to set
[iOSSim] Matched 1 Simulator cache item for cleanup: /Users/doomhammer/Library/Developer/CoreSimulator/Devices/8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8/data/Library/Caches/com.apple.mobile.installd.staging
[iOSSim] Setting common Simulator preferences to {"ConnectHardwareKeyboard":false}
[iOSSim] Updated 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8 Simulator preferences at '/Users/doomhammer/Library/Preferences/com.apple.iphonesimulator.plist' with {"ConnectHardwareKeyboard":false}
[iOSSim] The count of running Simulator UI client instances is 1
[iOSSim] Both Simulator with UDID 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8 and the UI client are currently running
[BaseDriver] Event 'simStarted' logged at 1517882768355 (10:06:08 GMT+0800 (+08))
[XCUITest] Reset requested. Removing app with id 'com.chopsosaurus.tableAndPicker' from the device
[XCUITest] Installing '/Users/doomhammer/Downloads/tableAndPicker.app' on Simulator with UUID '8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8'...
[XCUITest] The app has been installed successfully.
[BaseDriver] Event 'appInstalled' logged at 1517882771385 (10:06:11 GMT+0800 (+08))
[XCUITest] Using WDA path: '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent'
[XCUITest] Using WDA agent: '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'
[XCUITest] No obsolete cached processes from previous WDA sessions listening on port 8100 have been found
[JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[XCUITest] WDA is not listening at 'http://localhost:8100/'
[BaseDriver] Event 'wdaStartAttempted' logged at 1517882771552 (10:06:11 GMT+0800 (+08))
[XCUITest] Launching WebDriverAgent on the device
[XCUITest] Carthage found: '/usr/local/bin/carthage'
[XCUITest] Killing running processes 'xcodebuild.*8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8, iproxy 8100, 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8.*XCTRunner' for the device 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8...
[XCUITest] 'pgrep -nif xcodebuild.*8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8' didn't detect any matching processes. Return code: 1
[XCUITest] 'pgrep -nif iproxy 8100' didn't detect any matching processes. Return code: 1
[XCUITest] 'pgrep -nif 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8.*XCTRunner' didn't detect any matching processes. Return code: 1
[XCUITest] Using Xcode 9.2, so fixing WDA codebase
[BaseDriver] Event 'wdaStartFailed' logged at 1517882771707 (10:06:11 GMT+0800 (+08))
[XCUITest] Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
[XCUITest] Quitting and uninstalling WebDriverAgent, then retrying
[XCUITest] Shutting down sub-processes
[XCUITest] Removing WDA application from device
[BaseDriver] Event 'wdaStartAttempted' logged at 1517882782915 (10:06:22 GMT+0800 (+08))
[XCUITest] Launching WebDriverAgent on the device
[XCUITest] Carthage found: '/usr/local/bin/carthage'
[XCUITest] Killing running processes 'xcodebuild.*8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8, iproxy 8100, 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8.*XCTRunner' for the device 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8...
[XCUITest] 'pgrep -nif xcodebuild.*8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8' didn't detect any matching processes. Return code: 1
[XCUITest] 'pgrep -nif iproxy 8100' didn't detect any matching processes. Return code: 1
[XCUITest] 'pgrep -nif 8B321EE6-6CA8-4DE4-B5ED-1F6CEA3DCBD8.*XCTRunner' didn't detect any matching processes. Return code: 1
[XCUITest] Using Xcode 9.2, so fixing WDA codebase
[BaseDriver] Event 'wdaStartFailed' logged at 1517882783036 (10:06:23 GMT+0800 (+08))
[XCUITest] Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
[XCUITest] Quitting and uninstalling WebDriverAgent, then retrying
[XCUITest] Shutting down sub-processes
[XCUITest] Removing WDA application from device
[XCUITest] Error: Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
    at XCUITestDriver.quitAndUninstall$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/lib/driver.js:387:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at 
 Error: Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
    at XCUITestDriver.quitAndUninstall$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/lib/driver.js:387:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at 
[XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[iOSLog] Stopping iOS log capture
[MJSONWP] Encountered internal error running command: Error: Unable to launch WebDriverAgent because of xcodebuild failure: "ENOENT: no such file or directory, open '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/WebDriverAgent/PrivateHeaders/XCTest/XCUICoordinate.h'".
    at XCUITestDriver.quitAndUninstall$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/lib/driver.js:387:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at 
[HTTP] <-- POST /wd/hub/session 500 19568 ms - 384

**Comment by vutukuriswathi** [comment]
Even i am also getting the same error with 1.3.2 appium desktop

**Comment by zbranzov** [comment]
Same as @yotsubakoiwai . Does anyone investigate the issue?

**Comment by Swaagie** [comment]
Temporary fix https://github.com/facebook/WebDriverAgent/issues/840, you'll have to scroll all the way to the bottom for more details.

tl;dr

```
cd /Applications/Appium.app/Contents/Resources/app
npm install
```

**Comment by zbranzov** [comment]
Thanks @Swaagie

**Comment by dpgraham** [comment]
This has been resolved in latest version.
